pysmt/test/omt_examples.py

```diff
@@ -0,0 +1,551 @@
+        assert cost.is_real_constant() or cost.is_int_constant(), f"test: {test_id_str}, goal: {goal}, goal_value: {goal_value}, cost: {cost}"
+        goal_value_constant = goal_value.constant_value()
+        cost_constant = cost.constant_value()
+    pytest_test_case.assertEqual(
+        goal_value_constant,
+        cost_constant,
+        f"test_id: {test_id_str}, goal: {goal}, oracle value: {goal_value}, cost returned: {cost}"
+    )
+
+
+def _get_expected_raised_class(goals_value):
+    raised_class = None
+    if isinstance(goals_value, str):
+        if goals_value == "unbounded":
+            raised_class = PysmtUnboundedOptimizationError
+        elif goals_value == "infinitesimal":
+            raised_class = PysmtInfinitesimalError
+        else:
+            raise ValueError("Unknown value for goals_values")
+    return raised_class
+
+
+def _test_lexicographic(pytest_test_case, optimizer, goals, goals_values, test_id_str, is_sua_or_incr):
+    raised_class = _get_expected_raised_class(goals_values[0])
+    assert raised_class is None or len(goals_values) == 1, f"test: {test_id_str}, goals_values: {goals_values}"
+    if raised_class is None:
+        retval = optimizer.lexicographic_optimize(goals)
+        pytest_test_case.assertIsNotNone(retval, test_id_str)
+        _, costs = retval
+        assert len(goals) == len(goals_values) == len(costs), test_id_str
+        for goal, goal_value, cost in zip(goals, goals_values, costs):
+            _check_oracle_goal(pytest_test_case, goal, goal_value, cost, test_id_str)
+    elif not is_sua_or_incr:
+        with pytest_test_case.assertRaises(raised_class, msg=test_id_str):
+            optimizer.lexicographic_optimize(goals)
+
+
+def _test_pareto(pytest_test_case, optimizer, goals, goals_values, test_id_str, is_sua_or_incr):
+    raised_class = _get_expected_raised_class(goals_values[0])
+    assert raised_class is None or len(goals_values) == 1, f"test: {test_id_str}, goals_values: {goals_values}"
+    if raised_class is None:
+        retval = optimizer.pareto_optimize(goals)
+        pytest_test_case.assertIsNotNone(retval, test_id_str)
+
+        sorted_costs = sorted((costs for _, costs in retval), key=str)
+        sorted_goals_values = sorted(goals_values, key=str)
+
+        pytest_test_case.assertEqual(len(sorted_costs), len(sorted_goals_values), test_id_str)
+        for costs, goals_values in zip(sorted_costs, sorted_goals_values):
+            assert len(goals) == len(costs) == len(goals_values), test_id_str
+
+            for goal, goal_value, cost in zip(goals, goals_values, costs):
+                _check_oracle_goal(pytest_test_case, goal, goal_value, cost, test_id_str)
+    elif not is_sua_or_incr:
+        with pytest_test_case.assertRaises(raised_class, msg=test_id_str):
+            optimizer.pareto_optimize(goals)
+
+
+def _test_boxed(pytest_test_case, optimizer, goals, goals_values, test_id_str, test_basic, is_sua_or_incr):
```

Comment:
In general, I do not like that so much business logic of the tests is in this file (same for the other kinds of optimization). Why not put this in the `test_xxx` file(s)?

Reply:
Yes, but these are not part of the examples, they are more testing utils. Maybe we can put them in a separate file that is imported by the testing files? Something like `optimization_utils.py`?